### PR TITLE
cog init

### PR DESCRIPTION
**Pull request by zeke**
This PR introduces a new CLI command `cog init` that can be used to bootstrap an existing project with starter `cog.yaml` and `predict.py` files.

```
$ cog init
Setting up the current directory for use with Cog...
Created /Users/z/git/replicate/cog/cog.yaml
Created /Users/z/git/replicate/cog/predict.py
```

### To Do

- [x] Make sure I'm Doing It Right™, Golang-wise
- [x] Do not overwrite files if they already exist
- [x] Refine the `cog.yaml` and `predict.py` templates to contain the right starter content
- [x] Write docs
- [x] Write tests

### Stretch goals

- [ ] Derive `python_packages` somehow ([pipreqs](https://pypi.org/project/pipreqs/)? look for `requirements.txt`?)
- [ ] Add example `script/download-weights`?

### CODE REVIEW

**Comment by zeke** [comment]
This is ready for a first look! 👀

**Comment by bfirsh** [comment]
This looks fantastic!

> Derive python_packages somehow

We probably shouldn't put effort into this until we've decided on #157.

**Comment by zeke** [comment]
This now has tests and docs! It's ready for another look.

File: pkg/cli/init-templates/cog.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Configuration for Cog ⚙️
+# Reference: https://github.com/replicate/cog/blob/main/docs/yaml.md
+
+build:
+  # set to true if your model requires a GPU
+  gpu: false
+
+  # a list of ubuntu apt packages to install
+  system_packages:
+    - "libgl1-mesa-glx"
```
Comment (andreasjansson) [inline pkg/cli/init-templates/cog.yaml:10]: Might want some indication that these are just example packages. Maybe they should be commented out by default?

File: pkg/cli/init-templates/cog.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    - "libglib2.0-0"
+
+  # python version in the form '3.8' or '3.8.12'
```
Comment (andreasjansson) [inline pkg/cli/init-templates/cog.yaml:13]: Do patch versions work now? They didn't use to but perhaps @bfirsh added that.

File: pkg/cli/init-templates/cog.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  python_version: "3.8"
+
+  # a list of packages in the format <package-name>==<version>
+  python_packages:
+    - "numpy==1.19.4"
```
Comment (andreasjansson) [inline pkg/cli/init-templates/cog.yaml:18]: Same here, perhaps comment them out?

File: pkg/cli/init-templates/predict.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+	
+import cog
+# import torch
+
+class Predictor(cog.Predictor):
+    def setup(self):
+      """Load the model into memory to make running multiple predictions efficient"""
+      # self.model = torch.load("./weights.pth")
+
+    @cog.input("image", type=cog.Path, help="Grayscale input image")
```
Comment (andreasjansson) [inline pkg/cli/init-templates/predict.py:12]: Maybe we could have a comment pointing to the documentation for the predict() function here, to make it super easy to read about input types, etc.?

**Review by andreasjansson — APPROVED** [review]
Fantastic! This lowers the barrier to start using Cog by a lot!

File: pkg/cli/init-templates/cog.yaml (hunk 2 above)
Comment (bfirsh) [reply pkg/cli/init-templates/cog.yaml:13]: I think they always have worked? It just passes down to pyenv which supports either. The thing I added was just using Docker's `python` image, which also supports either.

File: pkg/cli/init-templates/predict.py (hunk 4 above)
Comment (zeke) [reply pkg/cli/init-templates/predict.py:12]: There's a comment at the top of the file:

```yaml
# Prediction interface for Cog ⚙️
# Reference: https://github.com/replicate/cog/blob/main/docs/python.md
```

File: pkg/cli/init-templates/cog.yaml (hunk 1 above)
Comment (zeke) [reply pkg/cli/init-templates/cog.yaml:10]: Yep we waffled on this. You tipped the scales. ⚖️

Comment (zeke) [reply pkg/cli/init-templates/cog.yaml]: 51f59536c135b97daa87676631ebe4b55e911f8e

File: pkg/cli/init-templates/cog.yaml (hunk 3 above)
Comment (zeke) [reply pkg/cli/init-templates/cog.yaml]: Yep. Done. 51f59536c135b97daa87676631ebe4b55e911f8e

File: pkg/cli/init-templates/cog.yaml (hunk 2 above)
Comment (zeke) [reply pkg/cli/init-templates/cog.yaml:13]: I just did a test `cog build` with `python_version: "3.8.11"` and it appears to work:

```
 => [internal] load metadata for docker.io/library/python:3.8.11                                                                        1.1s
 => [stage-0 1/8] FROM docker.io/library/python:3.8.11@sha256:a0a734233420b17d9ab37125afc9d8217b75db153d55854ac6683e639f00a8e8         13.4s
```

**Comment by zeke** [comment]
@bfirsh I made a few minor updates based on @andreasjansson's feedback and updated the output a bit. Here's what it looks like now:

```
Setting up the current directory for use with Cog...

✅ Created /Users/z/git/replicate/cog/cog.yaml
✅ Created /Users/z/git/replicate/cog/predict.py

Done! For next steps, check out the docs at https://github.com/replicate/cog/tree/main/docs
```

This is ready for another look before we ship it.

**Comment by bfirsh** [comment]
Looks like there are merge conflicts.

I wonder if this would be a better URL for docs? Actually describes what each thing is. https://github.com/replicate/cog#next-steps

Could even change the heading to "Documentation" so the anchor becomes `#documentation`.

**Review by bfirsh — APPROVED** [review]
Minor thing anyway. Looks good otherwise!

**Comment by zeke** [comment]
> I wonder if this would be a better URL for docs?

We discussed this in meatspace. Gonna ship with the docs URL as-is.

**Pull request merged**